### PR TITLE
Modify order_params method in frontend/orders_controller

### PR DESCRIPTION
**Pull request by aishmita-aggarwal**

### CODE REVIEW

File: frontend/spec/controllers/spree/orders_controller_spec.rb [hunk 1]
```diff
@@ -131,4 +126,40 @@
...
   end
+
+  describe '#order_params' do
+    let(:params) { controller.send(:params) }
+    let(:parameters) { ActionController::Parameters.new }
+    let(:order_params) { { order: { line_items_attributes: { "0" => { id: 1, quantity: 0 } } } } }
+
+    before do
+      allow(controller).to receive(:check_authorization)
+      allow(controller).to receive_messages(:current_order => order)
```
Comment (aishmita-aggarwal) [inline frontend/spec/controllers/spree/orders_controller_spec.rb]: Cannot use new hash syntax with string literal.

Comment (krzysiek1507) [inline frontend/spec/controllers/spree/orders_controller_spec.rb]:
```ruby
allow(controller).to receive_messages(:current_order => order)
```
@aishmita-aggarwal symbol ;)

**Comment by priyank-gupta** [comment]
:+1:

File: frontend/app/controllers/spree/orders_controller.rb [hunk 2]
```diff
@@ -95,11 +95,7 @@ def check_authorization
     private
 
       def order_params
-        if params[:order]
-          params[:order].permit(*permitted_order_attributes)
-        else
-          {}
-        end
+        params.fetch(:order, {}).permit(*permitted_order_attributes)
```
Comment (krzysiek1507) [inline frontend/app/controllers/spree/orders_controller.rb]:
```ruby
params.require(:order).permit(*permitted_order_attributes)
```

**Comment by Mafi88** [comment]
@aishmita-aggarwal Why every response in spec is mocked? Why do you ```expect(parameters).to receive(:permit)``` if it is not sent to controller?

**Comment by damianlegawiec** [comment]
This PR should be split into 2 separate ones:
1. One refactoring specs (removing duplications)
2. Another one with the change in `OrdersController`

**Pull request closed without merging**